macaroons/constraints.go

```diff
@@ -122,6 +150,39 @@ func IPLockChecker() (string, checkers.Func) {
 	}
 }
 
+// IPRangeLockChecker accepts client IP range from the validation context and
+// compares it with the IP range locked in the macaroon. It is of the `Checker`
+// type.
+func IPRangeLockChecker() (string, checkers.Func) {
+	return CondIpRange, func(ctx context.Context, cond, arg string) error {
+		// Get peer info and extract IP range from it for macaroon
+		// check.
+		pr, ok := peer.FromContext(ctx)
+		if !ok {
+			return fmt.Errorf("unable to get peer info from " +
```

Comment:
nit: use `errors.New` to avoid some bot-created PRs in the future to change it again 😂

Reply:
With 
```
return errors.New(fmt.Sprintf("incorrect macaroon IP range: %v", err))
```
the linter complains `should use fmt.Errorf(...) instead of errors.New(fmt.Sprintf(...)) (gosimple)`

Reply:
fixed